supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointParser.java

```diff
@@ -0,0 +1,216 @@
+
+        writeEndpointsToFile(endpointClasses);
+    }
+
+    /**
+     * Extracts endpoint information from the methods of a given class declaration.
+     *
+     * This method iterates over the methods of the provided class and their annotations.
+     * If an annotation matches one of the specified HTTP method annotations, it extracts
+     * the path values from the annotation and creates EndpointInformation objects for each path.
+     *
+     * @param javaClass                 the class declaration to extract endpoint information from
+     * @param httpMethodClasses         a set of HTTP method annotation class names
+     * @param classRequestMappingString the class-level request mapping string
+     * @return a list of EndpointInformation objects representing the extracted endpoint information
+     */
+    private static List<EndpointInformation> extractAnnotationPathValues(ClassOrInterfaceDeclaration javaClass, Set<String> httpMethodClasses, String classRequestMappingString) {
+        return javaClass.getMethods().stream().flatMap(method -> method.getAnnotations().stream().filter(annotation -> httpMethodClasses.contains(annotation.getNameAsString()))
+                .flatMap(annotation -> extractPathsFromAnnotation(annotation).stream()
+                        .map(path -> new EndpointInformation(classRequestMappingString, method.getNameAsString(), annotation.getNameAsString(), path, javaClass.getNameAsString(),
+                                method.getBegin().get().line, method.getAnnotations().stream().map(AnnotationExpr::toString).collect(Collectors.toList())))))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Extracts the paths from the given annotation.
+     *
+     * This method processes the provided annotation to extract path values.
+     * It handles both single-member and normal annotations, extracting the
+     * path values from the annotation's member values or pairs.
+     *
+     * @param annotation the annotation to extract paths from
+     * @return a list of extracted path values
+     */
+    private static List<String> extractPathsFromAnnotation(AnnotationExpr annotation) {
+        List<String> paths = new ArrayList<>();
+        if (annotation instanceof SingleMemberAnnotationExpr) {
+            Expression memberValue = ((SingleMemberAnnotationExpr) annotation).getMemberValue();
+            if (memberValue instanceof ArrayInitializerExpr) {
+                paths.addAll(((ArrayInitializerExpr) memberValue).getValues().stream().map(Expression::toString).collect(Collectors.toList()));
+            }
+            else {
+                paths.add(memberValue.toString());
+            }
+        }
+        else if (annotation instanceof NormalAnnotationExpr) {
+            ((NormalAnnotationExpr) annotation).getPairs().stream().filter(pair -> "value".equals(pair.getNameAsString())).forEach(pair -> paths.add(pair.getValue().toString()));
+        }
```

Comment:
Same here, you could use pattern matching for `instanceof` to get rid of the casts